tests/mlflow/tensorflow/test_tensorflow_model_export.py

```diff
@@ -55,32 +81,126 @@ def test_log_saved_model(self):
+                ("compression-ratio", [0.0]),
+                ("horsepower", [0.0]),
+                ("peak-rpm", [0.0]),
+                ("city-mpg", [0.0]),
+                ("highway-mpg", [0.0]),
+                ("price", [0.0])
+            ])
+
+            types = collections.OrderedDict((key, type(value[0]))
+                                            for key, value in defaults.items())
+            df = pandas.read_csv(path, names=types.keys(), dtype=types, na_values="?")
+            df = df.dropna()
+            # Keeping only the data we are testing.
+            df = df[["price", "body-style", "curb-weight", "highway-mpg"]]
+
+            # Extract the label from the features dataframe.
+            y_train = df.pop("price")
+
+            # Creating the input training function required.
+            trainingFeatures = {}
+
+            for i in df:
+                trainingFeatures[i] = df[i].values
+
+            input_train = tf.estimator.inputs.numpy_input_fn(trainingFeatures, 
+                                                            y_train.values, 
+                                                            shuffle=False, 
+                                                            batch_size=1)
+
+            # Creating the feature columns required for the DNNRegressor.
+            body_style_vocab = ["hardtop", "wagon", "sedan", "hatchback", "convertible"]
+            body_style = tf.feature_column.categorical_column_with_vocabulary_list(
+                key="body-style", vocabulary_list=body_style_vocab)
+            feature_columns = [
+            tf.feature_column.numeric_column(key="curb-weight"),
+            tf.feature_column.numeric_column(key="highway-mpg"),
+            # Since this is a DNN model, convert categorical columns from sparse
+            # to dense.
+            # Wrap them in an `indicator_column` to create a
+            # one-hot vector from the input.
+            tf.feature_column.indicator_column(body_style),]
+
+            # Build a DNNRegressor, with 2x20-unit hidden layers, with the feature columns
+            # defined above as input.
+            model = tf.estimator.DNNRegressor(
+                hidden_units=[20, 20], feature_columns=feature_columns)
+
+            # Training the model.
+            model.train(input_fn=input_train, steps=100)
+            # Saving the model's prediction on the training data.
+            model_predict = model.predict(input_train)
+            # Setting the logging such that it is in the temp folder and deleted after the test.
+            old_tracking_dir = tracking.get_tracking_uri()
+            tracking_dir = os.path.abspath(tmp.path("mlruns"))
+            tracking.set_tracking_uri("file://%s" % tracking_dir)
+            tracking.start_run()
+            try:
+                # Creating dict of features names (str) to placeholders (tensors)
+                feature_spec = {}
+                size = len(y_train.values)
```

Comment:
Nit: I don't think you need to specify the batch size explicitly here, do things work if you just do `shape=[None]` for each of the placeholders?